REGRESSION(298547@main): [ macOS wk2 ] http/tests/media/media-stream/get-display-media-prompt.html is a flaky text failure

youennf opened the pull request:
#### ed0423319e8814d304c9db4dfd3d6774476b069d
<pre>
REGRESSION(298547@main): [ macOS wk2 ] http/tests/media/media-stream/get-display-media-prompt.html is a flaky text failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=297411">https://bugs.webkit.org/show_bug.cgi?id=297411</a>
<a href="https://rdar.apple.com/158343702">rdar://158343702</a>

Reviewed by Charlie Wolfe.

The test change revealed a potential race condition between InjectedBundle::resetUserMediaPermissionRequestCount and InjectedBundle::userMediaPermissionRequestCount, since one is using synchronous IPC and not the other.
We are making use of synchronous IPC for both of them to remove that race.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::resetUserMediaPermissionRequestCount):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveSynchronousMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/298885@main">https://commits.webkit.org/298885@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5ea19cb62ebf1e69c7b0275797251d6ca1a9418a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116824 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36488 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122900 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67449 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4ad52825-4f6a-4454-8f24-536e609007f6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118713 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37186 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45089 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88712 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43145 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2ff46c69-a3b4-422c-924f-b8beabdc54e7) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119773 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29646 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104793 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69185 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28710 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22898 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66565 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99025 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23053 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126034 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43723 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32840 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97380 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44087 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100994 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97178 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24772 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42505 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20445 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39731 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43609 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43076 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46415 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44781 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe reviewed (approved):
You could probably use `postMessageWithAsyncReply` instead and await the promise it returns in the test, but this seems fine too.

youennf committed:
REGRESSION(298547@main): [ macOS wk2 ] http/tests/media/media-stream/get-display-media-prompt.html is a flaky text failure
https://bugs.webkit.org/show_bug.cgi?id=297411
rdar://158343702

Reviewed by Charlie Wolfe.

The test change revealed a potential race condition between InjectedBundle::resetUserMediaPermissionRequestCount and InjectedBundle::userMediaPermissionRequestCount, since one is using synchronous IPC and not the other.
We are making use of synchronous IPC for both of them to remove that race.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::resetUserMediaPermissionRequestCount):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveSynchronousMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/298885@main
